sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/EncodingPerFile.java

```diff
@@ -44,10 +44,11 @@ boolean encodingMatch(InputFile inputFile) {
     }
 
     // Case-Insensitive check
-    Charset roslynEncoding = globalReportProcessor.getRoslynEncodingPerUri().get(uri); 
+    Charset roslynEncoding = globalReportProcessor.getRoslynEncodingPerUri().get(uri);
     if (roslynEncoding == null) {
-      LOG.warn("File '{}' does not have encoding information. Skip it.", uri);
-      return false;
+      // Roslyn saw the file, but it didn't recognized it's encoding. We fall back to the default encoding and accept the file.
+      LOG.warn("Roslyn encoding was not found for '{}', using default instead.", uri);
```

Comment:
Wouldn't be more clear to write something like: "File encoding cannot be identified from Roslyn metadata for '{}', using default instead."

Reply:
`Roslyn metadata` seems ambiguous to me. It could refer to real Roslyn internal state or the `file-metadata.pb`.

I've tried to update the message to `Roslyn encoding was not detected for '{}', using default instead.`

Reply:
> Roslyn metadata seems ambiguous to me. It could refer to real Roslyn internal state or the file-metadata.pb

As far as I can tell, the Roslyn state (SyntaxTree.Encoding) is stored in `file-metadata.pb` and processed by this method so both are true :).

What about "Roslyn can not detect encoding for `{}`, using default instead."?